training/testing/test.py

```diff
@@ -130,26 +132,94 @@ def test_model(audio_test_path, label_path, mpath, model_arguments):
         )
         segment_ids_all.extend(list(zip(segment_ids, segment_ids_start_time.tolist())))
 
+    # collect all results in vectors
     gt_all_all = np.asarray(list(itertools.chain.from_iterable(gt_all)))
     preds_all_all = np.concatenate(list(itertools.chain.from_iterable(preds_all)))
     hr_all_all = np.asarray(list(itertools.chain.from_iterable(hr_all)))
     comments_all_all = np.asarray(list(itertools.chain.from_iterable(comments_all)))
 
-    np.save(identifier_string + "gt_all_all.npy", gt_all_all)
-    np.save(identifier_string + "preds_all_all.npy", preds_all_all)
-    np.save(identifier_string + "hr_all_all.npy", hr_all_all)
-    np.save(identifier_string + "comments_all_all.npy", comments_all_all)
-    np.save(identifier_string + "segment_ids_all.npy", np.asarray(segment_ids_all))
+    # save all results
+    np.save(os.path.join(results_path, "gt_all_all.npy"), gt_all_all)
+    np.save(
+        os.path.join(results_path, "preds_all_all.npy"),
+        preds_all_all,
+    )
+    np.save(os.path.join(results_path, "hr_all_all.npy"), hr_all_all)
+    np.save(
+        os.path.join(results_path, "comments_all_all.npy"),
+        comments_all_all,
+    )
+    np.save(
+        os.path.join(results_path, "segment_ids_all.npy"),
+        np.asarray(segment_ids_all),
+    )
 
 
 if __name__ == "__main__":
-    label_path = "/Data/test_dataset/labels"
-    audio_test_path = "/Data/test_dataset/audio"
-    mpath = "/app/assets/ckpt-epoch=21-val_loss=0.12-lr=0.005.ckpt"
-    identifier_string = "ben_model_"
-    model_arguments = {}
-
-    test_model(audio_test_path, label_path, mpath, model_arguments)
-
-
-    
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--label_path",
+        help="Path to the labels",
+        required=False,
+        default="/Data/test_dataset/labels",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--audio_path",
+        help="Path to the audio",
+        required=False,
+        default="/Data/test_dataset/audio",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--results_path",
+        help="Path to the directory where subdirectory with results will be created/overwritten",
+        required=False,
+        default="results/",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--model_path",
+        help="Path to the model weights",
+        required=False,
+        default="/app/assets/ckpt-epoch=21-val_loss=0.12-lr=0.005.ckpt",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--config",
+        help="Path to the config file, required if non-default model_arguments were used",
+        required=False,
+        type=str,
+    )
+
+    cli_args = parser.parse_args()
+    if not cli_args.config is None:
```

Comment:
I think you might be interested in using [Hydra](https://hydra.cc) ;-)

Reply:
Just going for consistency with the repository here :-P